### PR TITLE
This PR cleans up sys.rb for idiomatic problems

### PR DESCRIPTION
**Pull request by kernelsmith**
When attempting to normalize the behavior and return values (tho they are mostly meaningless for the cmd_ methods), I decided to add two constants to dispatcher_shell, namely COMMAND_SUCCESS and COMMAND_FAILURE.  Tho the return values on these are rarely checked, this gives a) an easy place to update them should they ever become important, b) a much clearer picture of the intent of the code and c) the start of the ability to use the return values for something.  This may help w/those folks who want to know when a command has finished etc.  

this code still needs more YARD, but I ran out of steam on that front.

To confirm, just run through the various sys commands at your leisure.

```
meterpreter > execute -f calc.exe
Process 4324 created.
meterpreter > kill 4324
Killing: 4324
meterpreter > getenv
[-] You must provide env vars to lookup, or -h

Usage: getenv varname [varname2] ...
meterpreter > getenv %COMSPEC%

Environment Variables
=====================

Variable  Value
--------  -----
COMSPEC   C:\Windows\system32\cmd.exe

meterpreter > getenv %COMSPEC% %COMPUTERNAME%

Environment Variables
=====================

Variable      Value
--------      -----
COMPUTERNAME  VICTIM
COMSPEC       C:\Windows\system32\cmd.exe

meterpreter > getpid
Current pid: 2512
meterpreter > getsid
Server SID: S-1-5-21-1720418662-3660741266-2307421305-1004
meterpreter > getuid
Server username: victim\victim
meterpreter > ps

Process List
============

 PID   PPID  Name                      Arch  Session  User              Path
 ---   ----  ----                      ----  -------  ----              ----
 0     0     [System Process]
 4     0     System
 296   4     smss.exe
 368   356   csrss.exe
 448   440   csrss.exe
 456   356   wininit.exe
 504   440   winlogon.exe
 552   456   services.exe
 560   456   lsass.exe
 572   456   lsm.exe
 660   552   svchost.exe
 724   552   vmacthlp.exe
 768   552   svchost.exe
...snip...

meterpreter > ps -h

Usage: ps [options]

Use the command with no arguments to see all running processes.
The following options can be used to filter those results:

OPTIONS:

    -A <opt>  Filters processes on architecture (x86 or x86_64)
    -S <opt>  String to search for (converts to regex)
    -U <opt>  Filters processes on the user using the supplied RegEx
    -h        Help menu.
    -s        Show only SYSTEM processes

meterpreter > ps -S cmd

Process List
============

 PID   PPID  Name                      Arch  Session  User              Path
 ---   ----  ----                      ----  -------  ----              ----
 4928  3528  cmd.exe                   x86   1        victim\victim  C:\Windows\system32\cmd.exe

meterpreter > reg

Usage: reg [command] [options]

Interact with the target machine's registry.

OPTIONS:

    -d <opt>  The data to store in the registry value.
    -h        Help menu.
    -k <opt>  The registry key path (E.g. HKLM\Software\Foo).
    -r <opt>  The remote machine name to connect to (with current process credentials
    -t <opt>  The registry value type (E.g. REG_SZ).
    -v <opt>  The registry value name (E.g. Stuff).
    -w        Set KEY_WOW64 flag, valid values [32|64].

COMMANDS:

    enumkey  Enumerate the supplied registry key [-k <key>]
    createkey  Create the supplied registry key  [-k <key>]
    deletekey  Delete the supplied registry key  [-k <key>]
    queryclass Queries the class of the supplied key [-k <key>]
    setval  Set a registry value [-k <key> -v <val> -d <data>]
    deleteval  Delete the supplied registry value [-k <key> -v <val>]
    queryval  Queries the data contents of a value [-k <key> -v <val>]

meterpreter > execute -f calc.exe
Process 2232 created.
meterpreter > suspend 2232
[*] Suspending: 2232
[*] Targeting process with PID 2232...
meterpreter > suspend -h

Usage: suspend [options] pid1 pid2 pid3 ...

Suspend one or more processes.

OPTIONS:

    -c        Continues suspending or resuming even if an error occurs
    -h        Help menu.
    -r        Resumes the target processes instead of suspending

meterpreter > suspend -r 2232
[*] Resuming: 2232
[*] Targeting process with PID 2232...
meterpreter > sysinfo
Computer        : VICTIM
OS              : Windows 7 (Build 7601, Service Pack 1).
Architecture    : x86
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
meterpreter >
```

### CODE REVIEW

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 1]
```diff
@@ -269,107 +292,175 @@ def cmd_shell(*args)
...
-    print_line("Server username: #{client.sys.config.getuid}")
+    if args.length > 0
+      cmd_getuid_help
+    else
+      print_line("Server username: #{client.sys.config.getuid}")
+    end
+  end
+
+  def cmd_getuid_help
+    print(%Q(
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: This adds inline newlines, please break it into multiple print_line() calls.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 2]
```diff
@@ -269,107 +292,175 @@ def cmd_shell(*args)
...
-    print_line("Server SID: #{client.sys.config.getsid}")
+    if args.length > 0
+      cmd_getsid_help
+    else
+      print_line("Server SID: #{client.sys.config.getsid}")
+    end
+  end
+
+  def cmd_getsid_help
+    print(%Q(
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line instead

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 3]
```diff
@@ -847,67 +1001,74 @@ def cmd_shutdown(*args)
...
   end
 
   #
   # help for the suspend command
   #
   def cmd_suspend_help
-    print_line("Usage: suspend [options] pid1 pid2 pid3 ...")
-    print_line("Suspend one or more processes.")
-    print @@suspend_opts.usage
+    print(%Q(
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line instead

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 4]
```diff
@@ -817,25 +964,32 @@ def cmd_sysinfo(*args)
...
+    end
+  end
 
-    client.sys.power.shutdown(force, SHTDN_REASON_DEFAULT)
+  #
+  # Help for the 'shutdown' command
+  #
+  def cmd_shutdown_help
+    print(%Q(
+Usage: shutdown [options]
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line instead

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 5]
```diff
@@ -751,49 +863,84 @@ def cmd_rev2self(*args)
...
-    print_line("")
+    COMMAND_SUCCESS
+  end
+
+  #
+  # Help for the 'getprivs' command
+  #
+  def cmd_getprivs_help
+    print_line(%Q(
+Usage: getprivs
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n, use print_line for each line.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 6]
```diff
@@ -587,161 +681,179 @@ def cmd_reg(*args)
...
+            open_key = remote_key.open_key(base_key, KEY_READ + wowflag)
           end
+        end
 
-          open_key.delete_value(value)
+        v = open_key.query_value(value)
 
-          print_line("Successfully deleted #{value}.")
+        print(%Q(
+Key: #{key}
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line for each line.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 7]
```diff
@@ -424,99 +515,116 @@ def validate_pids(pids, allow_pid_0 = false, allow_session_pid = false)
...
-    client.sys.power.reboot(force, SHTDN_REASON_DEFAULT)
+      client.sys.power.reboot(force, SHTDN_REASON_DEFAULT)
+    end
+  end
+
+  #
+  # Help for the 'reboot' command
+  #
+  def cmd_reboot_help
+    print(%Q(
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line for each line.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 8]
```diff
@@ -424,99 +515,116 @@ def validate_pids(pids, allow_pid_0 = false, allow_session_pid = false)
...
   def cmd_ps_help
-    print_line "Usage: ps [ options ]"
-    print_line
-    print_line "Use the command with no arguments to see all running processes."
-    print_line "The following options can be used to filter those results:"
-
-    print_line @@ps_opts.usage
+    print(%Q(
+Usage: ps [options]
+
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line for each line.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 9]
```diff
@@ -269,107 +292,175 @@ def cmd_shell(*args)
...
   #
   # help for the kill command
   #
   def cmd_kill_help
-    print_line("Usage: kill [pid1 [pid2 [pid3 ...]]] [-s]")
-    print_line("Terminate one or more processes.")
-    print_line(" -s : Kills the pid associated with the current session.")
+    print(%Q(
+Usage: kill [pid1 [pid2 [pid3 ...]]] [-s]
+
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line for each line.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb [hunk 10]
```diff
@@ -269,107 +292,175 @@ def cmd_shell(*args)
...
+        print_line
+        print_line(table.to_s)
+        COMMAND_SUCCESS
+      end
     end
   end
 
+  def cmd_getenv_help
+    print(%Q(
+Usage: getenv varname [varname2] ...
```
Comment (hmoore-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb]: Same as above, embedded \n in a print() call, use print_line for each line.

**Comment by hmoore-r7** [comment]
This is a lot of code churn for no practical improvement, what are the benefits of merging this?

**Comment by kernelsmith** [comment]
I thought the PR description was pretty clear. It's to bring the code in line with idiomatic guidelines, not to mention uncluster the completely haphazard code patterns/anti-patterns. Additionally, isn't the embedded newline thing kind of the point of using %Q?  Why are they so evil, do they break some UI?  Is t this better than all the hard-coded /n's like the code was littered with before I did this, even inside prints and print_line calls? This approach is far more readable not to mention getting rid of the ridiculous amount of repetitive calls to print_line

**Comment by hmoore-r7** [comment]
Embedded newlines aren't handled properly in various third-party GUIs, which is why we use print_line() instead of just embedding a "\n". As for the other changes, its not clear if the changes are an improvement in any way, and given the amount of review required to land this, they may not be worth the overhead.

**Comment by kernelsmith** [comment]
Aaand this is why the old shitty code will never get updated. I'm glad my 4 hours of effort were for naught

**Pull request closed without merging**